From 67be741dd5bb490493a38a6468dbca7c0782e101 Mon Sep 17 00:00:00 2001
From: Sarika Sharma <quic_sarishar@quicinc.com>
Date: Thu, 19 Sep 2024 14:37:54 +0530
Subject: [PATCH] wifi: ath12k: set flag for native/raw mode for ap monitor bringup

Currently, support for bringing AP and monitor on the same PHY is
not available.

Therefore, enable the flag SUPPORTS_CONC_AP_MON_IN_80211_FRAME for
native and raw mode.

Limitations:
For bringing Monitor + AP in same PHY-
Monitor mode:
RX MON connected clients packets will get dropped
TX MON connected clients data packets will get dropped.

patch-dependency: none

Signed-off-by: Sarika Sharma <quic_sarishar@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mac.c | 1 +
 1 file changed, 1 insertion(+)

--- a/drivers/net/wireless/ath/ath12k/mac.c
+++ b/drivers/net/wireless/ath/ath12k/mac.c
@@ -19375,6 +19375,8 @@ static int ath12k_mac_hw_register(struct
 
 		if (ath12k_ppe_ds_enabled)
 			ieee80211_hw_set(hw, SUPPORTS_VLAN_DATA_OFFLOAD);
+	} else {
+		ieee80211_hw_set(hw, SUPPORTS_CONC_AP_MON_IN_80211_FORMAT);
 	}
 
 	if (ab_dflt->hw_params->supports_ap_ps)
